specifyweb/frontend/js_src/lib/components/Forms/ResourceView.tsx

```diff
@@ -170,6 +174,36 @@ export function ResourceView<SCHEMA extends AnySchema>({
     viewName,
   });
 
+  const [openAsReadOnly] = userPreferences.use(
+    'form',
+    'behavior',
+    'openAsReadOnly'
+  );
+
+  const hasOwnButton =
+    !isDependent &&
+    !isSubForm &&
+    typeof resource === 'object' &&
+    formElement !== null;
+
+  const isNotReadOnlyContext = React.useContext(IsNotReadOnly);
```

Comment:
(optional) instead of creating a context (Which is a fine solution) you might be able to get away with just using `dialog === false` (i.e, only do readonly for forms opened not in a dialog)